server/src/main/java/org/elasticsearch/cluster/routing/RoutingChangesObserver.java

```diff
@@ -25,7 +25,7 @@ default void shardStarted(ShardRouting initializingShard, ShardRouting startedSh
     /**
      * Called when relocation of a started shard is initiated.
      */
-    default void relocationStarted(ShardRouting startedShard, ShardRouting targetRelocatingShard) {}
+    default void relocationStarted(ShardRouting startedShard, ShardRouting targetRelocatingShard, String reason) {}
```

Comment:
Does it makes sense to create an enum to represent the reason for relocation? There should be a finite amount of reasons. We can make the method more type safe and less error-prone for typos.

Reply:
Yeah, I think this is a good idea